### ISSUE TITLE
move() method eats the first email moved to new folder when create=True

### ISSUE THREAD

**Issue by jgrassler** [mailprocessing/mailprocessing#3]
tl;dr: maildirproc eats the first email moved to a new folder when creating folders automatically with `create=True`. Do not use this feature until this is fixed.

The `move()` method for IMAP emails has an optional `create` parameter that creates nonexistent folders. If that parameter is `True`, `imapproc` (`maildirproc` is unaffected) will

1. Create all non-existent folders and their parent folders if told to move a message to a non-existent folder
2. Fail to copy the message to the target folder
3. Delete the original message anyway.

The net result is that the first message to be moved to a non-existent folder will get deleted.

Please do not use the `create` parameter for now. I will fix this tonight (CEST).

**Comment by jgrassler** [comment]
Version 1.2.0 is affected, anything greater than that will have the fix. I removed 1.2.0 from pypi so this version will 404 rather than install on somebody's machine where it ends up eating emails.

**Comment by jgrassler** [comment]
This will look as follows in the log:

```
==> Moving UID 11253 to INBOX/lists/mylist
==> Copying 11253 to INBOX/lists/mylist
==> Destination folder INBOX/lists/mylist does not exist, creating.
==> Selecting folder INBOX
==> Folder INBOX selected.
==> Not creating folder ['INBOX']: folder exists.
==> Selecting folder INBOX
==> Folder INBOX selected.
==> Not creating folder ['INBOX', 'lists']: folder exists.
==> Selecting folder INBOX
==> Folder INBOX selected.
==> Creating folder INBOX/lists/mylist
==> Successfully created folder INBOX/lists/mylist
==> Deleting 11253
```

If you look for affected UIDs in `~/.mailprocessing/log-imap`, you will be able to at least recover some headers (`From:`, `To:`, `CC:`, `Date:`, `Subject:` and  `Message-ID:` are written to the log).

**Comment by jgrassler** [comment]
Fixed and tested in version `1.2.1` (available on PyPI, too).
